### DISCUSSION TITLE
Functions timeout after 10s!

### DISCUSSION THREAD

**Discussion by T04435** [vercel/community#236] [Help] [answered]
I have a project deployed on Vercel + NextJS.

In which I have a `function: /api/awesome/function` that does take a bit to complete from 30 - 120 seconds. All was fine on my localhost, but when I deployed into vercel :hankey: hit the FAN.

Function Logs in Vercel:
`2021-04-02T19:37:04.909Z [UUID] Task timed out after 10.01 seconds`

I see that this is due to my free plan resources: memory: 1GB  + maxDuration: 10s
Got docs from: https://vercel.com/docs/configuration#project/functions

Now the question is:
According to this doc: 

> Maximum Execution Duration: 10s (Hobby), 60s (Pro), or 900s (Enterprise)

My awesome-function could require me to go to an enterprise plan :cry: 

So is there anything, I can do here to not have to pay :moneybag: or at the worst case move to another stack? 

In this 10 seconds `awesome-function` processed 5 items and there are around 400-600

> In case anyone is curious about why `awesome-function` take that long
> - 20 x fetch data from an api => Array
> - processes the above Array
> - saves to AWS RDS

**Comment by CyberCyclone** [comment, marked as answer]
A bit late but here's a few options. 

1. Run all the tasks in parallel (Promise.all)
2. If it's a function that's running 24/7, Serverless is not the use for it. I'd recommend running a long processing task / job queue using the rpm package PM2. You can write a script that, for example, grabs 20 items, pass them to Promise.all, and once they're complete, terminate the script. PM2 will then rerun the script and it can grab the next 20. Or, you can have as many of these scripts running at once.
